tools/document-pii/src/item_collector.rs

```diff
@@ -0,0 +1,293 @@
+use std::collections::{BTreeMap, BTreeSet, HashMap};
+use std::fs;
+use std::io::BufRead;
+use std::path::{Path, PathBuf};
+
+use anyhow::anyhow;
+use syn::punctuated::Punctuated;
+use syn::visit::Visit;
+use syn::{ItemEnum, ItemStruct, UseTree};
+
+use crate::EnumOrStruct;
+
+/// Iterates over the rust files to collect all the types and use_statements, which is later
+/// used for recursively looking for pii_fields afterwards.
+pub struct AstItemCollector {
+    module_path: String,
+    /// Maps from the full path of a type to its AST node.
+    all_types: HashMap<String, EnumOrStruct>,
+    /// Maps from a module_path to all the use_statements and path of local items.
+    use_statements: BTreeMap<String, BTreeSet<String>>,
+}
+
+pub struct TypesAndUseStatements {
+    // Maps the path name of an item to its actual AST node.
+    pub all_types: HashMap<String, EnumOrStruct>,
+    // Maps the use_statements to different modules. For use in constructing the full path
+    // of an item from its type name.
+    pub use_statements: BTreeMap<String, BTreeSet<String>>,
+}
+
+impl AstItemCollector {
```

Comment:
Can we, please, put `impl` right after the struct definition (according our [`style guidelines`](https://develop.sentry.dev/rust/#declaration-order)) ?